tests/visualisation/resource_usage_visualisation_test.cpp

```diff
@@ -0,0 +1,24 @@
+#include "tests/common/sources/json_source.hpp"
+#include "tests/common/sinks/mock_sink.hpp"
+
+#include "visualisation/resource_usage_visualisation.hpp"
```

Comment:
We don't know what will done to the data, so 'Visualisation' is a bit misleading. Should it be something closer to, for example 'time series'